### PR TITLE
[7.3] CombineNestedDirnameFixer - support PHP 7.3

### PR DESCRIPTION
**Pull request by kubawerlos**
To `2.13` branch as the fixer was added in `2.13`.

Ping @gharlan (as original author) for review.

### CODE REVIEW

File: src/Fixer/FunctionNotation/CombineNestedDirnameFixer.php [hunk 1]
```diff
@@ -181,25 +181,43 @@ private function getDirnameInfo(Tokens $tokens, $index, $firstArgumentEndIndex =
...
 
-        $info['indexes'][] = $next;
         $info['secondArgument'] = $next;
         $info['levels'] = (int) $tokens[$next]->getContent();
 
         $next = $tokens->getNextMeaningfulToken($next);
 
-        if (!$tokens[$next]->equals(')')) {
-            return false;
+        if ($tokens[$next]->equals(')')) {
```
Comment (gharlan) [inline src/Fixer/FunctionNotation/CombineNestedDirnameFixer.php]: maybe move this after the `equals(',')` check?

```php
	        if ($tokens[$next]->equals(',')) {
	            $info['indexes'][] = $next;
	            $next = $tokens->getNextMeaningfulToken($next);
	        }

	        if (!$tokens[$next]->equals(')')) {
	            return false;
	        }

	        $info['indexes'][] = $next;
	        $info['end'] = $next;
	
	        return $info;
```

File: src/Fixer/FunctionNotation/CombineNestedDirnameFixer.php [hunk 2]
```diff
@@ -181,25 +181,43 @@ private function getDirnameInfo(Tokens $tokens, $index, $firstArgumentEndIndex =
         }
 
         $next = $tokens->getNextMeaningfulToken($next);
+        $info['indexes'][] = $next;
+
+        if ($tokens[$next]->equals(')')) {
```
Comment (gharlan) [inline src/Fixer/FunctionNotation/CombineNestedDirnameFixer.php]: I think, it is not necessary to duplicate this block.
Maybe:

```php
        if ($tokens[$next]->equals(',')) {
            $next = $tokens->getNextMeaningfulToken($next);
            $info['indexes'][] = $next;
        }

        if ($tokens[$next]->equals(')')) {
            $info['levels'] = 1;
            $info['end'] = $next;

            return $info;
        }

File: src/Fixer/FunctionNotation/CombineNestedDirnameFixer.php (hunk 1 above)
Comment (kubawerlos) [reply src/Fixer/FunctionNotation/CombineNestedDirnameFixer.php]: Cool, thanks for both comments - looks much better now.

**Comment by kubawerlos** [comment]
@gharlan anuthing more to improve?

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Comment by keradus** [comment]
I believe it's in good state, but if you spot sth @gharlan , don't hesitate !

**Pull request merged**